### PR TITLE
Prep for Solidus 4.0

### PR DESCRIPTION
**Pull request by cpfergus1**
This PR sets up this extension to work with Solidus 4.0+

### CODE REVIEW

File: solidus_jwt.gemspec [hunk 1]
```diff
@@ -30,7 +30,7 @@ Gem::Specification.new do |s|
 
   s.add_dependency 'jwt'
   s.add_dependency 'solidus_auth_devise'
-  s.add_dependency 'solidus_core', ['>= 2.0.0', '< 4']
```
Comment (kennyadsl) [inline solidus_jwt.gemspec:33]: why `>= 3.2.0`?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -18,7 +20,7 @@
 Dir[File.join(File.dirname(__FILE__), 'support/**/*.rb')].each { |f| require f }
 
 # Requires factories defined in lib/solidus_jwt/factories.rb
```
Comment (kennyadsl) [inline spec/spec_helper.rb]: Can you update the comment with the correct path? I think it's missing `/testing_support`

File: spec/lib/solidus_jwt/devise_strategies/refresh_token_spec.rb [hunk 3]
```diff
@@ -13,13 +13,11 @@
...
   end
 
-  let(:headers) { {} }
-  let(:user) { FactoryBot.create(:user, password: password) }
-  let(:password) { 'secret' }
+  let(:user) { FactoryBot.create(:user, password: 'secret') }
   let(:token) { user.auth_tokens.create! }
 
   before do
-    allow(request).to receive(:headers).and_return(:headers)
```
Comment (kennyadsl) [inline spec/lib/solidus_jwt/devise_strategies/refresh_token_spec.rb:23]: I don't understand this change, is `:headers` the same as `{}` in this context? Is this suggested by Rubocop?

**Review by kennyadsl — COMMENTED** [review]
Thanks! Left some comments here and there.

File: spec/lib/solidus_jwt/devise_strategies/refresh_token_spec.rb (hunk 3 above)
Comment (cpfergus1) [reply spec/lib/solidus_jwt/devise_strategies/refresh_token_spec.rb:23]: MultipleMemoizedHelpers
there were 7 in this spec so I had to try and reduce 2
Standardrb sounds pretty nice right now in place of Rubocop eh? 😄

File: solidus_jwt.gemspec (hunk 1 above)
Comment (cpfergus1) [reply solidus_jwt.gemspec:33]: 🤔 I guess for this one it doesn't matter, I will change it back

**Comment by kennyadsl** [comment]
Seems like [lint-code](https://app.circleci.com/pipelines/github/solidusio-contrib/solidus_jwt/255/workflows/52cb6c7f-7809-4983-ae69-6a414291e559/jobs/556) is failing. Can you take a look please?

**Pull request merged**